File: 백준/Bronze/1978. 소수 찾기/소수 찾기.js
const fs = require('fs');

// 소수 판별 함수 (효율적인 방법)
function isPrime(num) {
    if (num <= 1) return false;
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) return false;
    }
    return true;
}

// 입력을 읽어옵니다.
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const N = parseInt(input[0]);
const numbers = input[1].split(' ').map(Number);

// 소수의 개수를 셉니다.
let primeCount = 0;
for (let i = 0; i < N; i++) {
    if (isPrime(numbers[i])) {
        primeCount++;
    }
}

// 소수의 개수를 출력합니다.
console.log(primeCount);

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import sys

def is_prime(num):
    if num <= 1:
        return False
    for i in range(2,num):
        if num % i == 0:
            return False
    return True

N = int(sys.stdin.readline().strip())
numbers = list(map(int,sys.stdin.readline().strip().split()))

prime_count = 0
for number in numbers:
    if is_prime(number):
        prime_count += 1

print(prime_count)
    